arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (custom style)/VectorTileCustomStyleViewController.swift

```diff
@@ -17,40 +17,89 @@ import UIKit
-                           "1bf0cc4a4380468fbbff107e100f65a5"]
-    private var shownItemID: String? {
-        return ((mapView.map?.basemap.baseLayers.firstObject as? AGSArcGISVectorTiledLayer)?.item as? AGSPortalItem)?.itemID
+                           "1bf0cc4a4380468fbbff107e100f65a5",
+                           "2056bf1b350244d69c78e4f84d1ba215"]
+    
+    let temporaryURL: URL? = {
+        // Get a suitable directory to place files.
+        let directoryURL = FileManager.default.temporaryDirectory
+        // Create a unique name for the item resource cache based on current timestamp.
+        let formattedDate = ISO8601DateFormatter().string(from: Date())
+        // Create and return the full, unique URL.
+        return directoryURL.appendingPathComponent("\(formattedDate)")
+    }()
+    
+    // The item ID of the shown layer.
+    var shownItemID: String?
+    // The job to export the item resource cache.
+    var exportVectorTilesJob: AGSExportVectorTilesJob?
+    // The vector tiled layer created by the local VTPK.
+    var offlineVectorTiledLayer: AGSArcGISVectorTiledLayer?
+    
+    func loadOfflineLayer() {
+        // Get the vector tiled layer URL.
+        let vectorTiledLayerURL = URL(string: "https://arcgisruntime.maps.arcgis.com/home/item.html?id=2056bf1b350244d69c78e4f84d1ba215")!
+        // Create a vector tile cache using the local vector tile package.
+        let vectorTileCache = AGSVectorTileCache(name: "PSCC_vector")
+        // Create a portal item with the URL.
+        let portalItem = AGSPortalItem(url: vectorTiledLayerURL)!
+        // Create a task to export the custom style resources.
+        let task = AGSExportVectorTilesTask(portalItem: portalItem)
+        // Get the AGSExportVectorTilesJob.
+        exportVectorTilesJob = task.exportStyleResourceCacheJob(withDownloadDirectory: temporaryURL!)
+        // Start the job.
+        exportVectorTilesJob?.start(statusHandler: nil) { [weak self] (result, error) in
+            guard let self = self else { return }
+            if let result = result {
+                // Get the item resource cache from the result.
+                let itemresourceCahce = result.itemResourceCache
+                // Create a vector tiled layer with the vector tiled cache and the item resource cache.
+                self.offlineVectorTiledLayer = AGSArcGISVectorTiledLayer(vectorTileCache: vectorTileCache, itemResourceCache: itemresourceCahce)
+            } else if let error = error {
+                // Handle errors.
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    private func showSelectedItem(_ itemID: String) {
+        guard let map = mapView.map else { return }
+        shownItemID = itemID
+        if itemID == itemIDs.last {
+            // If the custom style is chosen, assign the local vector tile package to the basemap.
+            map.basemap = AGSBasemap(baseLayer: offlineVectorTiledLayer!)
```

Comment:
While it is usually OK to assume the offline layer is already loaded before user select it, this is not safe in the following situation:
- When a user has downloaded the on-demand resource, but lost network access afterwards. When they select the  offline custom style, the sample will crash.

Consider disabling the UI or blocking user interaction if the offlineVectorTiledLayer is nil.